src/battle_script_commands.c

```diff
@@ -9272,6 +9285,89 @@ static void Cmd_various(void)
             gBattlescriptCurrInstr += 7;
         return;
     }
+    case VARIOUS_JUMP_IF_FLING_FAILS:
```

Comment:
There's a placeholder function called `CanFling` which I feel it should be used instead of some of these checks, though it should be updated to remove checks already handled by
`if (!ItemId_GetFlingPower(gBattleMons[gActiveBattler].item))`.

It should be noted that there's already the check for Unnerve in there, so there's no need for the workaround in `attackcanceler` but it has to be fixed so it uses `IsUnnerveAbilityOnOpposingSide`.

Reply:
Better to use `UnnerveOn`, as that already has the check for the berry pocket.

Reply:
I'll look into it :eyes:

Reply:
> Better to use `UnnerveOn`, as that already has the check for the berry pocket.

Alright, so to clarify, by this you mean you want me to replace the `IsAbilityOnSide(battlerId, ABILITY_UNNERVE)` check inside of `CanFling` with `UnnerveOn(battlerId, item)` instead, correct? :eyes:

Reply:
> but it has to be fixed so it uses `IsUnnerveAbilityOnOpposingSide`.

I don't understand this part. It's an attacker with Unnerve who's not supposed to be able to use Fling if they're holding a berry, not an opponent 😅

Reply:
Source? Bulbapedia only says that Unnerve prevents flinging berries, and in every other instancUnnerve only affects the opposing side, so it makes sense that this would also affect the opposing side.

Reply:
> Source? Bulbapedia only says that Unnerve prevents flinging berries

No, it doesn't say "just" that 😅 

> Unnerve prevents a Pokémon holding a Berry from using Fling.

[That's specifically what it says](https://bulbapedia.bulbagarden.net/wiki/Fling_(move)).

Reply:
But it doesn't specify if the mon with Unnerve is the same one as the one holding the berry, it just says *a* Pokémon, not *which* one (itself, partner or opponents).
The phrasing is kinda off 🤔

Reply:
> But it doesn't specify if the mon with Unnerve is the same one as the one holding the berry, it just says _a_ Pokémon, not _which_ one (itself, partner or opponents). The phrasing is kinda off 🤔

It says quite clearly that a Pokémon with Unnerve holding a berry can't use Fling, which is a move. I don't know what's hard to understand about that, to be quite honest with you.